test/puppetlabs/puppetdb/http/index_test.clj

```diff
@@ -179,13 +190,13 @@
           (is (= results #{{:certname "host2"}})))))
 
     (testing "nodes"
-      (testing "query should return all nodes (including deactivated ones)"
+      (testing "query should return only active nodes"
         (doseq [query [["from" "nodes"]
                        "nodes {}"]]
           (is (= (set (mapv :certname (query-result method endpoint query {})))
                  #{"host1" "host2" "host3"}))))
 
-      (testing "broad regexp query should return all nodes"
+      (testing "broad regexp query should return all active nodes"
```

Comment:
I think for the new version, the way to go is probably to create a queryable pseudo-entity on nodes called "active" that defaults to true and represents deactivated + expired. This will make it easy for users to see deactivated nodes if they want, and also allow more fine-grained differentiation between deactivated and expired if needed via their respective columns.